mypy/messages.py

```diff
@@ -656,61 +657,53 @@ def too_few_arguments(self, callee: CallableType, context: Context,
+        msg = 'Missing named argument "{}"'.format(name) + for_function(callee)
         self.fail(msg, context)
 
     def too_many_arguments(self, callee: CallableType, context: Context) -> None:
-        msg = 'Too many arguments'
-        if callee.name:
-            msg += ' for {}'.format(callee.name)
+        msg = 'Too many arguments' + for_function(callee)
         self.fail(msg, context)
 
     def too_many_positional_arguments(self, callee: CallableType,
                                       context: Context) -> None:
-        msg = 'Too many positional arguments'
-        if callee.name:
-            msg += ' for {}'.format(callee.name)
+        msg = 'Too many positional arguments' + for_function(callee)
         self.fail(msg, context)
 
     def unexpected_keyword_argument(self, callee: CallableType, name: str,
                                     context: Context) -> None:
-        msg = 'Unexpected keyword argument "{}"'.format(name)
-        if callee.name:
-            msg += ' for {}'.format(callee.name)
+        msg = 'Unexpected keyword argument "{}"'.format(name) + for_function(callee)
         self.fail(msg, context)
         module = find_defining_module(self.modules, callee)
         if module:
             assert callee.definition is not None
-            self.note('{} defined here'.format(callee.name), callee.definition,
+            self.note('{} defined here'.format(callable_name(callee)), callee.definition,
                       file=module.path, origin=context)
 
     def duplicate_argument_value(self, callee: CallableType, index: int,
                                  context: Context) -> None:
         self.fail('{} gets multiple values for keyword argument "{}"'.
-                  format(capitalize(callable_name(callee)),
-                         callee.arg_names[index]), context)
+                  format(callable_name(callee) or 'Function', callee.arg_names[index]),
+                  context)
 
-    def does_not_return_value(self, callee_type: Optional[Type], context: Context) -> None:
+    def does_not_return_value(self, callee_type: FunctionLike, context: Context) -> None:
         """Report an error about use of an unusable type."""
-        name = None  # type: Optional[str]
-        if isinstance(callee_type, FunctionLike):
-            name = callee_type.get_name()
-        if name is not None:
-            self.fail('{} does not return a value'.format(capitalize(name)), context)
-        else:
-            self.fail('Function does not return a value', context)
+        name = callable_name(callee_type) or 'Function'
+        self.fail('{} does not return a value'.format(name), context)
+
+    def evaluates_to_none(self, context: Context) -> None:
+        self.fail('Expression evaluates to "None"', context)
+
+    def yield_from_iterator(self, context: Context) -> None:
+        self.fail('Expression "yield from <iterator>" does not return a value', context)
```

Comment:
This part is not directly related to the PR as I understand. So that it is better to make a separate PR/issue.
These error messages deserve a separate discussion.

Reply:
I thought I needed this change, but turns out you're right. Reverted.

(Although I do like the new messages better :) )